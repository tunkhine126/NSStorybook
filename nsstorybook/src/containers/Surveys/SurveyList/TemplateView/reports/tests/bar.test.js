/**
 *
 * Tests for Bar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';

import Bar from '../bar';

describe('<Bar />', () => {
  const props = {
    answer: 'Test',
    numberOfResponses: 25,
    percentage: 30,
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <Bar {...props} />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <Bar {...props} />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
