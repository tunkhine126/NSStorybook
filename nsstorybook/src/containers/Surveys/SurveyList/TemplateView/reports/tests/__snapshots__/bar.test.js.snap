// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-container-145 MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    class="MuiGrid-root makeStyles-percentageContainer-146 MuiGrid-item"
  >
    <p
      class="MuiTypography-root makeStyles-percentageText-147 MuiTypography-body1"
    >
      30
      %
    </p>
  </div>
  <div
    class="MuiGrid-root makeStyles-barContainer-144 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-true"
  >
    <div
      class="makeStyles-percentageBar-148"
    />
    <div
      class="MuiGrid-root makeStyles-barTextContainer-143 MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
    >
      <div
        class="MuiGrid-root makeStyles-responseContainer-149"
      >
        <p
          class="MuiTypography-root makeStyles-responseText-150 MuiTypography-body1"
        >
          Test
        </p>
      </div>
      <div
        class="MuiGrid-root"
      >
        <p
          class="MuiTypography-root makeStyles-numberOfResponses-151 MuiTypography-body1"
        >
          25
           
          Responses
        </p>
      </div>
    </div>
  </div>
</div>
`;
